src/slider/stateful-slider-container.js

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import React from 'react';
+import {STATE_CHANGE_TYPE} from './constants';
+import type {
+  StatefulContainerPropsT,
+  StateReducerT,
+  StateT,
+  ParamsT,
+  ChangeActionT,
+} from './types';
+
+const defaultStateReducer: StateReducerT = (type, nextState) => nextState;
+
+class StatefulSliderContainer extends React.Component<
+  StatefulContainerPropsT,
+  StateT,
+> {
+  static defaultProps = {
+    initialState: {},
+    stateReducer: defaultStateReducer,
+    onChange: () => {},
+  };
+
+  state = {...this.props.initialState};
+
+  onChange = (config: {event: *, params: ParamsT}) => {
+    const {event, ...params} = config;
+    //$FlowFixMe
+    this.internalSetState(STATE_CHANGE_TYPE.value, event, params);
+    const {onChange} = this.props;
+    return onChange({event, ...params});
```

Comment:
Isn't it going to produce args that look like next:
```
{
  event,
  params: {
    value: 'some value'
  }
}
```
Based on [this change](https://github.com/uber-web/baseui/pull/372/files) we don't next extra arguments under `params` prop, but it should look like next:
```
{
  event,
  value: 'some value'
}
```

Reply:
I checked 
![image](https://user-images.githubusercontent.com/1851469/47396176-9a067b80-d6de-11e8-8757-35b5766ce243.png)

Reply:
Well, I would `console.log` the arguments the `onChange` handler gets instead of creating a custom case like in the screenshot you posted above.
<img width="616" alt="screen shot 2018-10-25 at 4 09 49 pm" src="https://user-images.githubusercontent.com/1920399/47535438-80953900-d870-11e8-9637-51c7fb10d74b.png">

Reply:
I added a comment based on the type on line 32, but your console.log clearifies that the args look  fine. It looks like just the type of the config on line 32 is incorrect `onChange = (config: {event: *, params: ParamsT}) => {`. Should be `onChange = (config: {event: *, value: some_value_type}) => {`. Unless I'm missing something.